src/rules/spaceInParensRule.ts

```diff
@@ -0,0 +1,234 @@
+      if (this.exceptionsArrayOptions.length) {
+        this.braceException = this.exceptionsArrayOptions.indexOf('{}') !== -1;
+        this.bracketException = this.exceptionsArrayOptions.indexOf('[]') !== -1;
+        this.parenException = this.exceptionsArrayOptions.indexOf('()') !== -1;
+        this.empty = this.exceptionsArrayOptions.indexOf('empty') !== -1;
+      }
+    }
+  }
+
+  private getExceptions() {
+    const openers = [];
+    const closers = [];
+
+    if (this.braceException) {
+      openers.push(ts.SyntaxKind.OpenBraceToken);
+      closers.push(ts.SyntaxKind.CloseBraceToken);
+    }
+
+    if (this.bracketException) {
+      openers.push(ts.SyntaxKind.OpenBracketToken);
+      closers.push(ts.SyntaxKind.CloseBracketToken);
+    }
+
+    if (this.parenException) {
+      openers.push(ts.SyntaxKind.OpenParenToken);
+      closers.push(ts.SyntaxKind.CloseParenToken);
+    }
+
+    if (this.empty) {
+      openers.push(ts.SyntaxKind.CloseParenToken);
+      closers.push(ts.SyntaxKind.OpenParenToken);
+    }
+
+    return {
+      openers,
+      closers
+    };
+  }
+
+  protected findParenNodes(node: ts.Node): ts.Node[] {
+    const children = node.getChildren();
+    let first: ts.Node;
+    let second: ts.Node;
+    let penultimate: ts.Node;
+    let last: ts.Node;
+    for (let i = 0; i < children.length; i++) {
+      if (children[i].kind === ts.SyntaxKind.OpenParenToken) {
+        first = children[i];
+        second = children[i + 1];
+      }
+      if (children[i].kind === ts.SyntaxKind.CloseParenToken) {
+        penultimate = children[i - 1];
+        last = children[i];
+      }
+    }
+    return [ first, second, penultimate, last ];
+  }
+
+  protected visitNode(node: ts.Node): void {
+    const parenNodes = this.findParenNodes(node);
+    this.checkParanSpace(parenNodes[0] , parenNodes[1] , parenNodes[2], parenNodes[3]);
+    super.visitNode(node);
+  }
+
+  private checkParanSpace(first: ts.Node , second: ts.Node, penultimate: ts.Node, last: ts.Node) {
+    if (!first && !second && !penultimate && !last) return;
+
+    if (this.shouldOpenerHaveSpace(first, second)) {
+      const fix = this.createFix(this.appendText(first.getEnd(), ' '));
+      this.addFailure(this.createFailure(first.getEnd(), 0, Rule.MISSING_SPACE_MESSAGE, fix));
+    }
+    if (this.shouldOpenerRejectSpace(first, second)) {
+      const fix = this.createFix(this.deleteText(first.getEnd(), 1));
+      this.addFailure(this.createFailure(first.getEnd(), 0 , Rule.REJECTED_SPACE_MESSAGE, fix));
+    }
+    if (this.shouldCloserHaveSpace(penultimate, last)) {
+      const fix = this.createFix(this.appendText(penultimate.getEnd(), ' '));
+      this.addFailure(this.createFailure(last.getStart() , 0, Rule.MISSING_SPACE_MESSAGE, fix));
+    }
+    if (this.shouldCloserRejectSpace(penultimate, last)) {
+      const fix = this.createFix(this.deleteText(penultimate.getEnd(), 1));
```

Comment:
In the tests lets force one of them to have at least two spaces to delete. Right now this fixer will only remove one space.